storage/Utils/SystemCmd.cc

```diff
@@ -561,6 +617,34 @@ namespace storage
     }
 
 
+    void
+    SystemCmd::sendStdin()
+    {
+        if ( ! _childStdin )
+            return;
+
+        if ( ! _stdinText.empty() )
+        {
+            string::size_type count = 0;
+            string::size_type len   = _stdinText.size();
+            int result = 1;
+
+            while ( count < len && result > 0 )
+                result = fputc( _stdinText[ count++ ], _childStdin );
```

Comment:
fputc() is buffered I/O, i.e. it's buffered on the C stdin library level; it collects data until the underlying buffer is full and then sends that data off with a single write() call. The idea is to decouple the higher-level I/O from the application side from the low-level system buffer sizes. That's what the stdin lib does for us.